core/src/main/java/org/elasticsearch/ingest/core/ConfigurationUtils.java

```diff
@@ -79,6 +79,20 @@ private static String readString(String processorType, String processorTag, Stri
         throw newConfigurationException(processorType, processorTag, propertyName, "property isn't a string, but of type [" + value.getClass().getName() + "]");
     }
 
+    public static Integer readIntProperty(String processorType, String processorTag, Map<String, Object> configuration, String propertyName, Integer defaultValue) {
+        Object value = configuration.remove(propertyName);
+        if (value == null && defaultValue != null) {
+            return defaultValue;
+        } else if (value == null) {
+            return null;
```

Comment:
Can we remove this else if statement? In the case of the attachment processor we never return null. We can then also change the return type from `Integer` to `int` and we can do the same for the type of the default value.